### PR TITLE
Add Postgres source

### PR DESCRIPTION
**Pull request by danlamanna**
@mgrauer No action associated with this source, it also stealth fixes a bug where jade would flip out if your source had no icon/action.

Very simple source, just meant to be used as an input to analyses.

### CODE REVIEW

File: plugin_tests/postgres_test.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            'name': name,
+            'username': username,
+            'password': password,
+            'dbname': dbname,
+            'baseURL': baseURL
+        }
+        response = self.request(path=path, method='POST', params=params, user=self._user)
+        self.assertStatusOk(response)
+        postgresSource = response.json
+        print(response.json)
```
Comment (mgrauer) [inline plugin_tests/postgres_test.py]: can you clean this out if it isn't desired?

File: web_external/templates/body/sourcePanel.jade [hunk 2]
```diff
@@ -10,9 +10,11 @@
...
                         - var attributes = {'cid': source.cid}
                         - var sourceType = sourceTypes[source.getSourceType()]
-                        - var classes = 'm-icon-enabled m-display-source-action ' + sourceType.icon + ' ' + sourceType.action
-                        i(class=classes)&attributes(attributes)
-                        //- trash icon to delete source
+
+                        if sourceType
+                          - var classes = 'm-icon-enabled m-display-source-action ' + sourceType.icon + ' ' + sourceType.action
+                          i(class=classes)&attributes(attributes)
+                          //- trash icon to delete source
```
Comment (mgrauer) [inline web_external/templates/body/sourcePanel.jade]: unindent this comment, or better yet, remove it as it is self-evident.

**Comment by mgrauer** [comment]
Is this a WIP, or all you wanted to do here?  It doesn't actually create a postgres connection of any kind.

**Comment by mgrauer** [comment]
>it also stealth fixes a bug where jade would flip out if your source had no icon/action.

the first rule of stealth fixes...

:speak_no_evil:

File: web_external/js/views/widgets/AddPostgresSourceWidget.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            }, this).createSource(params);
+        }
+    },
+
+    initialize: function (settings) {
+        this.collection = settings.collection;
+    },
+
+    render: function () {
+        var modal = this.$el.html(minerva.templates.addPostgresSourceWidget({}));
```
Comment (mgrauer) [inline web_external/js/views/widgets/AddPostgresSourceWidget.js]: call `girderModal`, see the note in the dev guide, currently in #210.

**Comment by danlamanna** [comment]
It doesn't create a connection, if you want me to include ```psycopg2``` and setup some basic functionality for executing a query from user input or something, I can. However this is what I had remaining from #122, where a PG source was just used as input to an analysis.

**Comment by mgrauer** [comment]
No need to do that.  I was curious about how much you intended to add with this PR.

**Comment by danlamanna** [comment]
Sure - this is set on my end.

**Comment by mgrauer** [comment]
I made three changes.

1) Rebased on master

2)Reverted the girderModal, b/c this was causing a flash b/w the first dialog (add source) and the next (add postgres source).  I'll need to clarify when to use girderModal in the docs, but I think it is better not to in this case because we have already called girderModal on the first dialog, then the container just gets re-used by the second dialog.

3) Changed the icon to the recently added 'icon-database'.

This branch LGTM if these changes are OK by you.

**Comment by danlamanna** [comment]
image

**Pull request merged**